src/ol/renderer/canvas/ImageLayer.js

```diff
@@ -211,7 +227,46 @@ class CanvasImageLayerRenderer extends CanvasLayerRenderer {
       }
     }
 
-    const img = this.getImage();
+    /**
+     * @type {HTMLImageElement|HTMLCanvasElement|HTMLVideoElement|ImageBitmap}
+     */
+    let img = image.getImage();
+    if (img !== this.previousImg_ && this.imageBitmap_) {
+      this.imageBitmap_.close();
+      this.imageBitmap_ = null;
+    }
+    if (
+      img === this.previousImg_ &&
+      !this.getLayer().getSource()?.scaleable &&
+      !(img instanceof HTMLCanvasElement) &&
+      img.width * img.height < 268435456
```

Comment:
Seem worth adding a comment with a link to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas#maximum_canvas_size or naming this in a way that makes it less of a mysterious number.